corehq/motech/repeaters/static/repeaters/js/repeat_record_report.js

```diff
@@ -240,5 +238,21 @@ hqDefine('repeaters/js/repeat_record_report', function () {
                 },
             });
         }
+
+        function setAction(action) {
+            window.action = action;
```

Comment:
Attaching things to `window` isn't a good practice, it's just another form of globals. I still think this UI would be less unwieldy in knockout, but as it is, since you need access to `action` in the `#confirm-button` handler, I'd store it as a data attribute on the confirm button. It's basically treating the confirm button as a small form with a hidden "action" input.